flow/scheduling/slurm.py

```diff
@@ -140,9 +140,7 @@ def submit(
         submit_cmd = self.submit_cmd + flags
 
         if after is not None:
-            submit_cmd.extend(
-                ["-W", 'depend="afterany:{}"'.format(after.split(".")[0])]
-            )
+            submit_cmd.extend(["-W", "-d", "afterany:{}".format(after.split(".")[0])])
```

Comment:
Why do we need to split the string when there is a period? From the docs, it looks like a list of job ids is listed like `-d afterany:00000:00001:00002` etc...

Reply:
The `after` argument is largely unused AFAIK and at various points in flow's history it has definitely been outright broken. This logic is >5 years old and has not been changed in that time (I can definitely trace it back to March 2016 at least) and has survived many large changes in flow, so it's possible that this split was necessary at one point. `s.split('.')[0]` is a no-op if the string doesn't contain `'.'`, so it's not causing any harm, but it's probably safe to remove.

Reply:
I agree with @vyasr that this is probably legacy code that may not apply here. I'd remove the `split` because it's confusing and probably not helpful.
```suggestion
            submit_cmd.extend(["-W", "-d", "afterany:{}".format(after)])
```